app/src/main/java/http_server/Router.java

```diff
@@ -27,14 +27,24 @@ public String routeRequest(RequestReader request) {
 
       case "/head_request":
         if ("HEAD".equals(method)) {
-          return handler.handleHeadRequest();
+          return handler.handleHead();
+        }
+        return handler.handleNonHead();
+
+      case "/method_options":
+        if ("OPTIONS".equals(method)) {
+          return handler.handleMethodOptions("allow", "GET, HEAD, OPTIONS");
+        }
+
+      case "/method_options2":
+        if ("OPTIONS".equals(method)) {
+          return handler.handleMethodOptions("allow", "GET, HEAD, OPTIONS, PUT, POST");
```

Comment:
You mentioned the flow doesn't make sense. Perhaps your Router shouldn't be building the response. I would think about what is a router supposed to be doing. Perhaps something else builds the response. Perhaps the communication flow needs rethinking. Would it help to take a step back and think about the information flow and see from a bigger picture standpoint?

This is advice for team dynamics and expectations when working on code bases, creating PRs:
The one comment I have about your PR description is that you really don't want to wait until you put up a final PR before you bring up concerns. What is better to do is:
- You starting a slack convo in the channel and getting feedback from your team in a thread to get a better understanding. If we were on the same time zone, I would want you to reach out to me directly since I started the convo and saying the concerns and seeing if we could chat about it further so we could review it together and figure out a plan or give you clarifications.   
- You putting up a WIP PR that shows the preliminary changes and saying that you wanted to start that conversation this way to show what the code would look like and see what direction should be taken. 

Does that make sense? The first thing I thought when I saw the PR was "why didn't she reach out and ask further" so I want you to get into the habit of understanding how to work in a team environment. The final PR is a compilation of your research, information gathering (from internet, other people) and consolidation of that into your code. If this isn't a final PR then its better to mention that in the PR title as WIP and that info in the description.

Reply:
Oh okay. yeah that totally makes sense. thank you :) I'll make sure to do this in the future. 

On this pr I'll reach out on slack about this issue as well, since I think it's looking like it's going to be a couple days long issue.